shared/src/main/scala/drt/shared/Api.scala

```diff
@@ -752,14 +764,26 @@ object CrunchApi {
     val waitTime: Int
   }
 
-  case class DeskStat(desks: Option[Int], waitTime: Option[Int])
+  case class DeskStat(desks: Option[Int], waitTime: Option[Int]) extends MinuteComparison[CrunchMinute] {
+    override def maybeUpdated(existing: CrunchMinute, now: MillisSinceEpoch): Option[CrunchMinute] =
+      if (existing.actDesks != desks || existing.actWait != waitTime) Option(existing.copy(
+        actDesks = desks, actWait = waitTime, lastUpdated = Option(now)
+      ))
+      else None
+  }
 
   case class ActualDeskStats(portDeskSlots: IMap[String, IMap[String, IMap[MillisSinceEpoch, DeskStat]]]) extends PortStateMinutes {
     val oneMinuteMillis: Long = 60 * 1000
 
     def applyTo(portState: PortStateMutable, now: MillisSinceEpoch): PortStateDiff = {
       val crunchMinutesDiff = minutes.foldLeft(List[CrunchMinute]()) { case (soFar, (key, dm)) =>
-        mergeMinute(key, portState.crunchMinutes.getByKey(key), dm, now) :: soFar
+        portState.crunchMinutes.getByKey(key) match {
+          case None => CrunchMinute(key, dm).copy(lastUpdated = Option(now)) :: soFar
+          case Some(existing) => dm.maybeUpdated(existing, now) match {
+            case None => soFar
+            case Some(updated) => updated :: soFar
+          }
+        }
```

Comment:
this code block is reused in several places, perhaps it's worth splitting into a function.